### PR TITLE
Make shelley `TxWits` consistent with other eras

### LINKED ISSUE #3959: Make `TxWits` from `Cardano.Ledger.Shelley.TxWits` consistent with other eras
Definition of witness in Shelley is very outdated and inconsistent with newer eras. We need to fix that. Some of the things that need to be done:

* Remove usage of `HKD`
* Switch to using `MemoBytes`

In other words make it look and feel similar to  `Cardano.Ledger.Alonzo.TxWits`

### PR DESCRIPTION
**Pull request by Lucsanszky**
# Description
Resolves #3959

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

**Comment by Lucsanszky** [comment]
I also moved `mapTraverseableDecoderA` from `Alonzo.TxWits` to `Shelley.TxWits`. Although, it might be better to move it to `Decoding` in `cardano-ledger-binary`. Any suggestions?

I'll also update the `CHANGELOG`s and then this will be ready for review.

**Review by lehins — COMMENTED** [review]
Looking good.

> I also moved mapTraverseableDecoderA from Alonzo.TxWits to Shelley.TxWits. 

That's totally fine


> Although, it might be better to move it to Decoding in cardano-ledger-binary. Any suggestions?

In not so distant future we will be getting rid of `Annotator`, so we don't really need to worry about the ideal placement for this function.

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs [hunk 1]
```diff
@@ -249,23 +220,26 @@ shelleyEqTxWitsRaw txWits1 txWits2 =
...
 --     Uses the non-exported WitnessSet' constructor.
 prettyWitnessSetParts ::
+  EraScript era =>
   ShelleyTxWits era ->
   ( Set (WitVKey 'Witness (EraCrypto era))
   , Map (ScriptHash (EraCrypto era)) (Script era)
   , Set (BootstrapWitness (EraCrypto era))
   )
-prettyWitnessSetParts (ShelleyTxWitsConstr (WitnessSet' a b c _)) = (a, b, c)
+prettyWitnessSetParts (ShelleyTxWits a b c) = (a, b, c)
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs]: I think we can remove this function and just use the `ShelleyTxWits` or lenses in the place where it is being used in the tests

**Review by lehins — APPROVED** [review]
Perfect! :+1:

**Pull request merged**